sinch/core/clients/sinch_client_configuration.py

```diff
@@ -58,6 +65,35 @@ def __init__(
         else:
             self.logger = logging.getLogger("Sinch")
 
+    def _set_sms_origin_with_service_plan_id(self):
+        self._sms_origin_with_service_plan_id = self._sms_domain_with_service_plan_id.format(self._sms_region)
+
+    def _set_sms_region_with_service_plan_id(self, region):
+        self._sms_region = region
+        self._set_sms_origin_with_service_plan_id()
+
+    def _get_sms_region_with_service_plan_id(self):
+        return self._sms_region
+
+    sms_region_with_service_plan_id = property(
+        _get_sms_region_with_service_plan_id,
+        _set_sms_region_with_service_plan_id,
+        doc="SMS Region for service plan id version of the SMS API"
+    )
+
+    def _set_sms_domain_with_service_plan_id(self, domain):
+        self._sms_domain = domain
+        self._set_sms_origin()
```

Comment:
IIRC we have customers that are using custom endpoints for SMS, so the mechanism is legit.

For testing environments:

`sinch_client.configuration.sms_origin = "..."`

Reply:
This is not what I meant: this purpose of this function is to override the default SMS domain, which is exactly that the first line is doing.
But then, calling `self._set_sms_origin()` seems unnecessary as this method will reformat the overridden domain with the region. Do you confirm you expect users to use parameterized  domains as overrides?

Reply:
Yes, I'll have to add that to the docs (Docs Team will be notified). Parametrised domains are not necessary.
In case of passing a domain without Python's f string parameter, the format function will not return error, but just return string without region (valid case).

In case of testing environments and customers that want to skip this logic entirely and provide sms origin that was constructed on their own, all that needs to be done is:

`sinch_client.configuration.sms_origin = "..."`
or
`sinch_client.configuration.sms_origin_with_service_plan_id = "..."`

I'll also add that to the docs.